spec/services/prisoner_slot_availability_spec.rb

```diff
@@ -0,0 +1,112 @@
+require "rails_helper"
+
+RSpec.describe PrisonerSlotAvailability do
+  let(:prison)        { create(:prison) }
+  let(:offender_id)   { 'A1410AE' }
+  let(:date_of_birth) { '1960-06-01' }
+  let(:start_date)    { Date.parse('2017-02-10') }
+  let(:end_date)      { Date.parse('2017-04-11') }
+
+  subject do
+    described_class.new(prison, offender_id, date_of_birth, start_date..end_date)
+  end
+
+  def all_slots_available
+    {
+      ConcreteSlot.new(2017, 2, 13, 14, 0, 16, 10).to_s => [],
```

Comment:
Can we make the test more concise by removing some of the slots? I don't think we need so many slots to test this class